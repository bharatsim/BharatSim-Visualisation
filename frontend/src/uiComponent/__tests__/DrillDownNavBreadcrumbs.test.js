import React from 'react';
import { fireEvent } from '@testing-library/dom';
import { render } from '@testing-library/react';
import DropDownNavBreadcrumbs from '../DrillDownNavBreadcrumbs';

describe('<DrillDownNavBreadcrumbs />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <DropDownNavBreadcrumbs
        items={[
          { label: 'top level', onClick: jest.fn() },
          { label: 'level 2', onClick: jest.fn() },
          { label: 'level 3', onClick: jest.fn() },
        ]}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should not create button for last element', () => {
    render(
      <DropDownNavBreadcrumbs
        items={[
          { label: 'top level', onClick: jest.fn() },
          { label: 'level 2', onClick: jest.fn() },
          { label: 'level 3', onClick: jest.fn() },
        ]}
      />,
    );

    const allItems = document.getElementsByTagName('li');

    expect(allItems[allItems.length - 1]).toMatchSnapshot();
  });

  it('should called callback on click on clickable list item', () => {
    const callback = jest.fn();
    const { getByText } = render(
      <DropDownNavBreadcrumbs
        items={[
          { label: 'top level', onClick: callback },
          { label: 'level 2', onClick: jest.fn() },
          { label: 'level 3', onClick: jest.fn() },
        ]}
      />,
    );

    const topLevelButton = getByText('top level');
    fireEvent.click(topLevelButton);

    expect(callback).toHaveBeenCalled();
  });
});
