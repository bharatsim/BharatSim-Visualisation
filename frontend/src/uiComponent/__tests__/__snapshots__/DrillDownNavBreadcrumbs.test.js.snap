// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrillDownNavBreadcrumbs /> should match snapshot 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
    class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    style="z-index: 23;"
  >
    <ol
      class="MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text WithStyles(ForwardRef(Button))-text-1 MuiButton-textSizeSmall MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            top level
          </span>
        </button>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text WithStyles(ForwardRef(Button))-text-1 MuiButton-textSizeSmall MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            level 2
          </span>
        </button>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 WithStyles(ForwardRef(Typography))-subtitle2-2"
        >
          level 3
        </h6>
      </li>
    </ol>
  </nav>
</div>
`;

exports[`<DrillDownNavBreadcrumbs /> should not create button for last element 1`] = `
<li
  class="MuiBreadcrumbs-li"
>
  <h6
    class="MuiTypography-root MuiTypography-subtitle2 WithStyles(ForwardRef(Typography))-subtitle2-4"
  >
    level 3
  </h6>
</li>
`;
